halo2_frontend/src/circuit.rs

```diff
@@ -112,6 +112,9 @@ pub fn compile_circuit<F: Field, ConcreteCircuit: Circuit<F>>(
     let selector_polys = selectors_to_fixed.convert(assembly.selectors);
     fixed.extend(selector_polys);
 
+    // sort the "copies" for deterministic ordering
+    assembly.permutation.copies.sort_by(|a, b| a.0.cmp(&b.0));
```

Comment:
This sort only considers the left hand side of each copy pair.  So the order will not be stable.  For example both (a-b, a-c) and (a-c, a-b) would be sorted according to this sorting.
I would suggest just doing this to sort comparing the entire pair (tuple):
```rust
assembly.permutation.copies.sort();
```

Reply:
Done https://github.com/privacy-scaling-explorations/halo2/pull/296/commits/c039c2927dd77ec2d73c0997a8681a25a300944e